server/routes/updateChainEntityTitle.ts

```diff
@@ -0,0 +1,52 @@
+import { Request, Response, NextFunction } from 'express';
+import { Op } from 'sequelize';
+import proposalIdToEntity from '../util/proposalIdToEntity';
+import lookupCommunityIsVisibleToUser from '../util/lookupCommunityIsVisibleToUser';
+
+export const Errors = {
+  NoEntity: 'Cannot find entity',
+  NotAdminOrOwner: 'Not an admin or owner of this entity',
+};
+
+const updateChainEntityTitle = async (models, req: Request, res: Response, next: NextFunction) => {
+  const [chain, community, error] = await lookupCommunityIsVisibleToUser(models, req.body, req.user);
+  if (error) return next(new Error(error));
+  const { unique_id, title } = req.body;
+
+  const entity = await proposalIdToEntity(models, chain.id, unique_id);
+  if (!entity) return next(new Error(Errors.NoEntity));
+  const userOwnedAddressIds = await req.user.getAddresses().filter((addr) => {
+    return !!addr.verified;
+  }).map((addr) => addr.id);
+  // Todo: author check
+  // if (!userOwnedAddressIds.includes(entity.address_id)) {
+  const roles = await models.Role.findAll({
+    where: {
+      address_id: { [Op.in]: userOwnedAddressIds, },
+      permission: { [Op.in]: ['admin', 'moderator'] },
+    }
+  });
+  const role = roles.find((r) => {
+    return r.offchain_community_id === entity.community || r.chain_id === entity.chain;
+  });
+  if (!role) return next(new Error(Errors.NotAdminOrOwner));
+  // }
+
+  entity.title = title;
+  await entity.save();
+
+  const finalEntity = await models.ChainEntity.findOne({
```

Comment:
We don't need to re-fetch here assuming that `entity` contains all the included models we need. Can just return `entity.toJSON()`.

Reply:
This query includes a ChainEvent and the default findEntity doesn't so...I'm gonna leave it in for now. But maybe it can be removed in a later iteration.